src/Autocomplete/src/Doctrine/EntityMetadata.php

```diff
@@ -42,21 +42,65 @@ public function isEmbeddedClassProperty(string $propertyName): bool
     }
 
     public function getPropertyMetadata(string $propertyName): array
+    {
+        trigger_deprecation('symfony/ux-autocomplete', '2.15.0', 'Calling EntityMetadata::getPropertyMetadata() is deprecated. You should stop using it, as it will be removed in the future.');
+
+        try {
+            return $this->getFieldMetadata($propertyName);
+        } catch (\InvalidArgumentException $e) {
+            return $this->getAssociationMetadata($propertyName);
+        }
+    }
+
+    /**
+     * @internal
+     *
+     * @return array<string, mixed>
+     */
+    public function getFieldMetadata(string $propertyName): array
     {
         if (\array_key_exists($propertyName, $this->metadata->fieldMappings)) {
-            return $this->metadata->fieldMappings[$propertyName];
+            $metadata = $this->metadata->fieldMappings[$propertyName];
+
+            // In doctrine/orm:^3.0; $metadata will be a FieldMapping object
+            if (!\is_array($metadata)) {
+                return (array) $metadata;
+            }
```

Comment:
For ORM 3, won't this return `[0 => FieldMapping]`? But in 2.9, it would be an associative array containing the metadata? Or am I misunderstanding?

Reply:
Yes in ORM 3, the `ClassMetaData::$fieldMappings` is of type `array<string, FieldMapping>` ([see](https://github.com/doctrine/orm/blob/3.0.x/src/Mapping/ClassMetadata.php#L334-L336)) and in ORM 2 it's just an `array` (`mixed` in docs, [see](https://github.com/doctrine/orm/blob/2.10.x/lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php#L449-L452)), it's been defined as a `psalm-type` ([see](https://github.com/doctrine/orm/blob/2.10.x/lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php#L73-L92)), that may cause the confusion?

Since we access the `ClassMetaData::$fieldMappings` with a 'key' it will return a `FieldMapping` object directly and casting it to an array works for the `FieldMapping` object as it is defined using public properties.